### DISCUSSION TITLE
how to create a diff.zip file , i want upload a file cover a old ,and create new history version

### DISCUSSION THREAD

**Discussion by probelover** [ONLYOFFICE/DocumentServer#1933] [Q&A]
How to generate a diff.zip ，please tell me Source code location

**Comment by ShockwaveNN** [comment]
Hi, sorry, I don't think there is any easy way to generate diff.zip file, since it calculated between two object in memory

So only way to do it I think dig up in code, but I think this is very complicated task

**Reply by probelover** [reply]
Yes, can you tell me the location of the code，the code in documentserver sources  ?

**Reply by ShockwaveNN** [reply]
It's located somewhere in [server](https://github.com/ONLYOFFICE/server) repo, but you need to find exact places by yourself

**Reply by probelover** [reply]
thanks
